M2/Macaulay2/d/actors3.d

```diff
@@ -1647,7 +1647,7 @@ map(e:Expr):Expr := (
 	  then map(a.0,a.1,a.2)
 	  else WrongNumArgs(2,3))
      else WrongNumArgs(2,3));
-setupfun("apply",map);
+setupfun("apply",map).Protected = false;
```

Comment:
Given how inefficient methods are implemented in the top level, I would be very hesitant about this change. I'd rather `scan` and `apply` remained low level functions until methods are made more efficient.

Reply:
Perhaps a new `iterate` method can play that role for now.

Reply:
Good point.  I'll play around with this some more.  Perhaps I can move the `scan` and `apply` code for iterators to the interpreter for the best of both worlds.